### PR TITLE
simplify tests

### PR DESCRIPTION
**Pull request by matklad**
No more write_mode everywhere :)

### CODE REVIEW

File: tests/system.rs [hunk 1]
```diff
@@ -297,29 +290,20 @@ fn handle_result(result: HashMap<String, String>,
...
-            (Path::new("tests/coverage-source/"), "tests/coverage-target/")
+fn get_target(file_name: &str, target: Option<&str>) -> String {
+    if file_name.contains("source") {
+        let target_file_name = file_name.replace("source", "target");
+        if let Some(replace_name) = target {
+            Path::new(&target_file_name)
+                .with_file_name(replace_name)
+                .into_os_string()
+                .into_string()
+                .unwrap()
```
Comment (matklad) [inline tests/system.rs]: This if branch does not work for all possible file names because it manipulates strings and not Paths, but I think this OK: we are not evil enough to create edge cases for ourselves =)

File: tests/writemode/source/fn-single-line.rs [hunk 2]
```diff
@@ -0,0 +1,80 @@
+// rustfmt-fn_single_line: true
+// rustfmt-write_mode: checkstyle
```
Comment (matklad) [inline tests/writemode/source/fn-single-line.rs]: This is a copy of usual test case file with `write-mode` option added

**Comment by nrc** [comment]
Thanks!
